contracts/koh/contracts/KingAutomaton.sol

```diff
@@ -324,66 +301,247 @@ contract KingAutomaton {
     uint256 mask = ((1 << bitsPerVote) - 1) << offset;
 
     // Get vote
-    return (ballotBoxes[_id].votes[index] & mask) >> offset;
+    return (self.ballotBoxes[_id].votes[index] & mask) >> offset;
+  }
+
+  function completeProposal(Data storage self, uint256 _id) public {
+    require(self.ballotBoxes[_id].state == BallotBoxState.Active, "Ballot is not active!");
+    Proposal storage p = self.proposals[_id];
+    ProposalState p_state = p.state;
+    require(p_state == ProposalState.Started || p_state == ProposalState.Contested, "Invalid proposal state!");
+    p.state = ProposalState.Completed;
+    self.ballotBoxes[_id].state = BallotBoxState.Inactive;
   }
+}
+
+contract KingAutomaton {
 
-  function getVoteCount(uint256 _id, uint256 _choice) public view
-      validBallotBoxID(_id) returns(uint256) {
-    return ballotBoxes[_id].voteCount[_choice] & ALL_BUT_MSB;
+  int256 constant INT256_MIN = int256(uint256(1) << 255);
+  int256 constant INT256_MAX = int256(~(uint256(1) << 255));
+  uint256 constant UINT256_MIN = 0;
+  uint256 constant UINT256_MAX = ~uint256(0);
+  uint256 constant MSB_SET = 1 << 255;
+  uint256 constant ALL_BUT_MSB = MSB_SET - 1;
+
+  using Miner for Miner.Data;
+  using Proposals for Proposals.Data;
+  using DEX for DEX.Data;
+  //////////////////////////////////////////////////////////////////////////////////////////////////
+  // Initialization
+  //////////////////////////////////////////////////////////////////////////////////////////////////
+
+  constructor(uint256 nSlots, uint256 minDifficultyBits, uint256 predefinedMask, uint256 initialDailySupply,
+      int256 approval_pct, int256 contest_pct, uint256 treasury_limit_pct) public {
+    numSlots = nSlots;
+    miner_data.initMining(nSlots, minDifficultyBits, predefinedMask, initialDailySupply);
+    initNames();
+
+    require(approval_pct > contest_pct, "Approval percentage must be bigger than contest percentage!");
+    proposals_data.approvalPercentage = approval_pct;
+    proposals_data.contestPercentage = contest_pct;
+    proposals_data.treasuryLimitPercentage = treasury_limit_pct;
+    // Check if we're on a testnet (We will not using predefined mask when going live)
+    if (predefinedMask != 0) {
+      // If so, fund the owner for debugging purposes.
+      debugging = true;
+      mint(msg.sender, 1000000 ether);
+      balances[treasuryAddress] = 1000000 ether;
+    }
+  }
+
+  //////////////////////////////////////////////////////////////////////////////////////////////////
+  // ERC20 Token
```

Comment:
I would extract the ERC20 functionality as an abstract base class for readability.